src/Objs/Redirect/RedirectComponent.tsx

```diff
@@ -0,0 +1,40 @@
+import * as React from 'react'
+import {
+  ensureUserIsLoggedIn,
+  isUserLoggedIn,
+  provideComponent,
+  urlFor,
+} from 'scrivito'
+import { InPlaceEditingPlaceholder } from '../../Components/InPlaceEditingPlaceholder'
+import { Redirect } from './RedirectObjClass'
+
+provideComponent(Redirect, ({ page }) => {
+  const requireLogin = page.get('requireLogin')
+  const link = page.get('link')
+  const url = link && urlFor(link)
+
+  React.useEffect(() => {
+    if (requireLogin && !isUserLoggedIn) {
```

Comment:
If I'm not mistaken, this is a function, not a constant:

```suggestion
    if (requireLogin && !isUserLoggedIn()) {
```